### ISSUE TITLE
Test suite hangs on some systems

### ISSUE THREAD

**Issue by eserte** [atrodo/Net-Dropbear#4]
On some of my smokers the test suite hangs. Here's a full test log until it was killed:
```
Output from '/usr/bin/make test':

"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Dropbear.bs blib/arch/auto/Net/Dropbear/Dropbear.bs 644
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/000-basic.t ............. ok
[26894] Dec 05 07:13:18 Failed loading /etc/dropbear/dropbear_rsa_host_key
[26894] Dec 05 07:13:18 Failed loading /etc/dropbear/dropbear_dss_host_key
[26894] Dec 05 07:13:18 Failed loading /etc/dropbear/dropbear_ecdsa_host_key
[26894] Dec 05 07:13:18 Failed listening on '22': Error listening: Permission denied
[26894] Dec 05 07:13:18 Not backgrounding
[26894] Dec 05 07:13:18 Early exit: Terminated by signal
t/010-sshd.t .............. ok
t/011-log.t ............... ok
t/012-username.t .......... ok
t/013-passwd_fill.t ....... ok
t/014-shadow_fill.t ....... ok
t/015-crypt_password.t .... ok
Terminated
make: *** [Makefile:1048: test_dynamic] Error 143
```

This seems to happen on newer linux systems (fedora 28, Ubuntu 18.04).

**Comment by eserte** [comment]
Still happens with Net-Dropbear-0.12.

**Comment by atrodo** [comment]
Thank you for the confirmation that it is still happening, I suspected it was but with the snafu that was 0.11 I hadn't been sure yet. Having finally gotten access to an Ubuntu 18.04 machine, I discovered that it's an update to openssh's output that caused this to happen. The ssh command now outputs what host (and port at least in this case) the authentication error originated from. That wasn't making my tests happy.

So I've done 2 things. First, I noticed that the script would get into an endless select cycle and reset the alarm, preventing the alarm from triggering and test failing. I cleaned up how my tests were running and hopefully the stalling will stop. Second I made the search for the Permission denied no longer anchored to the beginning of line to help this test match in more cases. Both of those things should end up resolving this in 0.13 that I plan on releasing soon.
